avocado/core/parser.py

```diff
@@ -80,4 +81,16 @@ def finish(self):
 
         Side effect: set the final value for attribute `args`.
         """
-        self.args = self.application.parse_args(namespace=self.args)
+        self.args, _ = self.application.parse_known_args(namespace=self.args)
+        if _:
+            for sub in self.application._subparsers._actions:
+                if sub.dest == 'subcommand':
+                    sub.choices[self.args.subcommand].print_usage()
+                    # Respecting the argparse exit code on unknown arg.
+                    # Using exit_codes.AVOCADO_JOB_FAIL here seems just wrong.
+                    sys.exit(2)
```

Comment:
I'm not sure argparse's "magic" number 2 is supposed to be something applications (or users) should depend on. I noticed that we define/describe:

```
#: Something else went wrong and avocado failed (or crashed). Commonly
#: used on command line validation errors.
AVOCADO_FAIL = 3
```

And it looks like it's more important to support our own users with a properly defined/documented error code. @ldoktor @lmr what do you guys think?